src/dbg/debugger.cpp

```diff
@@ -612,6 +613,31 @@ void cbMemoryBreakpoint(void* ExceptionAddress)
     cbGenericBreakpoint(BPMEMORY, ExceptionAddress);
 }
 
+void cbRunToUserCodeBreakpoint(void* ExceptionAddress)
+{
+    EXCLUSIVE_ACQUIRE(LockRunToUserCode);

```

Comment:
Won't this deadlock if you try to go to user code when waiting inside this callback?